appveyor.yml

```diff
@@ -107,12 +107,22 @@ for:
     - cd cmake_build
     - cmake
       -L
+      -DWARNINGS_FATAL=ON
+      -DDEBUG_ASSERTIONS_FATAL=ON
```

Comment:
Note that we will need to disable this when the we start to use CI for deployment. Maybe we enable conditionally for PRs but not for master builds/deployments.

Reply:
Added TODO comments